### ISSUE TITLE
Anonymous Class parentheses check for PER 2.0

### ISSUE THREAD

**Issue by rikvdh** [PHPCSStandards/PHP_CodeSniffer#568]
For PER 2.0 we need a sniff to check the absence of parentheses when no arguments are needed for an anonymous class.

> 8. Anonymous Classes[¶](https://www.php-fig.org/per/coding-style/#8-anonymous-classes)
> 
> ...
>
> If the anonymous class has no arguments, the () after class MUST be omitted.

The current existing sniff for 'anonymous classes' (`PSR12.Classes.AnonClassDeclaration`) could be extended with an configuration option. But maybe this specific behavior is easily checked in a new Sniff as well.

- [X] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [ ] I intend to create a pull request to implement this feature.

**Comment by jrfnl** [comment]
@rikvdh Please add a code sample of the desired behaviour.

**Comment by rikvdh** [comment]
This is invalid, the parentheses are not allowed there:
```php
<?php

$foo = new class() {};
```

Valid should be:
```php
<?php

$foo = new class {};
```

This is currently not covered by any sniff.

**Comment by jrfnl** [comment]
Confirmed for PHPCS. For what it's worth, PHPCSExtra does contain a sniff for this: [`Universal.Classes.DisallowAnonClassParentheses`](https://github.com/phpcsstandards/phpcsextra?tab=readme-ov-file#universalclassesdisallowanonclassparentheses-wrench-bar_chart-books).

**Comment by rikvdh** [comment]
Ok, I've overlooked this one. We'll go with this one for now.. 👍
